feat(erc20): add indexes at genesis state to track active token pairs

zsystm opened the pull request:
## Description

### Naming Change
The following maps exist for the purpose of indexing to find `TokenPair`. I believe the new names better reflect their purpose:
- `DenomMap` --> `TokenPairDenomIndex`
- `ERC20Map` --> `TokenPairERC20AddressIndex`

Aligned the following method names with their original intent and actual actions:
- `GetDenomMap` --> `GetTokenPairIdByDenom`
- `GetERC20Map` --> `GetTokenPairIdByERC20Addr`

### Added indexes as state

**Problem**
- [The original InitGenesis logic](https://github.com/Canto-Network/Canto/blob/v7.0.1/x/erc20/genesis.go#L26-L31) iterates over all existing TokenPair to set denomIndex(= `denomMap`) and erc20AddressIndex(= `erc20Map`).
- The problem arises when multiple token pairs exist for same denom. The iteration order is based on bytes, not the actual registration time.
- This means the logic might not correctly identify the most recently registered TokenPair.
- For example, the denomIndex for `ibc/13B6057538B93225F6EBACCB64574C49B2C1568C5AE6CCFE0A039D7DAC02BF29` ends up with the TokenPair associated with `0x2C68D1d6aB986Ff4640b51e1F14C716a076E44C4`, which is incorrect.
- The correct TokenPair should be the one associated with `0xD32eB974468ed767338533842D2D4Cc90B9BAb46`, as it was registered later and is currently active.

**Solution**
- Added indexes at genesis state to track active token pair. Now we can track the latest(= active) one if there were multiple token pairs at the same denom.

### Notes
- Actual logic fix will be handled as another PR.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] targeted the correct branch (see [PR Targeting](https://github.com/canto-network/canto/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] included the necessary unit and integration tests
- [x] reviewed "Files changed" and left comments if necessary
 
### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

zsystm committed:
feat: add indexes as genesis state

- multiple token pairs can exists at the same denom
- but only one active token pair can exist by index
- so index must be handled as genesis state. without it, we cannot track what is the active token pair from multiple token pairs for same denom

zsystm committed:
tmp: disable genesis validation

zsystm committed:
chore: update keyname

zsystm committed:
chore: revert Makefile change

zsystm committed:
chore: add comments

dudong2 commented on x/erc20/genesis_test.go:
```diff
@@ -82,30 +129,35 @@ func (suite *GenesisTestSuite) TestERC20InitGenesis() {
 	for _, tc := range testCases {
-
+		// TODO: fix duplicate problem and uncomment this line
+		//suite.Nil(tc.genesisState.Validate(), "genesis state should be valid")
```
no need to uncomment?

zsystm replied:
@dudong2 
I removed it. Thanks.

zsystm committed:
fix: import genesis logic

problem
- one token pair has two indexes
  - erc20 index
  - denom index
- so, set token pair per indexes and delete keys is wrong
  - when iterate erc20 indexes, id is already gone. so it will set empty token pair multiple times

solution
- set token pair once when iterate token pairs

zsystm committed:
chore: fix comment

zsystm committed:
chore: remove comments

- currently we already have multiple token pairs for same denom which is not allowed for validation genesis

- we have not decided yet for handling this.

zsystm committed:
fix: wrong test cases

Regardless of whether it’s OWNER_EXTERNAL or OWNER_MODULE, both indexes are being managed.

tkkwon1998 commented:
lgtm!

dongsam commented on x/erc20/genesis_test.go:
```diff
@@ -21,6 +22,66 @@ import (
+		},
+		{
+			Denom: uqstars,
+			// denomIdx must have latest token pair id
```
The token pair id is a hash, not a number, but currently, "latest" might be misinterpreted as numerically largest. How about modifying it as suggested below?

```suggestion
			// denomIdx must have the latest token pair id assigned
```

dongsam reviewed (approved):
LGTM, apart from the chore suggestion on the comment

dudong2 reviewed (approved):
lgtm

zsystm committed:
Update x/erc20/genesis_test.go

Co-authored-by: dongsam <dongsamb@gmail.com>
